homeassistant/components/edl21/__init__.py

```diff
@@ -1 +1,32 @@
 """The edl21 component."""
+
+from homeassistant import config_entries, core
+from homeassistant.const import Platform
+from .sensor import DOMAIN
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+) -> bool:
+    """Set up platform from a ConfigEntry."""
+    # Forward the setup to the sensor platform.
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, Platform.SENSOR)
+    )
+    return True
+
+
+async def async_unload_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+) -> bool:
+    """Unload a config entry."""
+    edl = hass.data[DOMAIN]
+
+    # Disconnect
+    await edl.disconnect()
```

Comment:
Shouldn't we first unload the plaform (and thus their entities) before we disconnect?